cmake/external/eigen.cmake

```diff
@@ -5,5 +5,8 @@ if (onnxruntime_USE_PREINSTALLED_EIGEN)
     file(TO_CMAKE_PATH ${eigen_SOURCE_PATH} eigen_INCLUDE_DIRS)
     target_include_directories(eigen INTERFACE ${eigen_INCLUDE_DIRS})
 else ()
+    execute_process(COMMAND  git apply --ignore-space-change --ignore-whitespace ${PROJECT_SOURCE_DIR}/patches/eigen/Fix_Eigen_Build_Break.patch
```

Comment:
is this only needed on ARM? Wondering why we aren't encountering this.

Reply:
We had the same question :) There are a couple of GCC compiler flags like "-Wunused-parameter" and "-Wno-unused-parameter" used in various places in cmakes. We could not identity which of them targets the external eigen component, they depends on a lot of factors: compiler, platform, if the component is prebuilt, etc:
https://github.com/microsoft/onnxruntime/blob/v0.5.1/cmake/CMakeLists.txt#L434
BTW, a similar compiler flag is used for oxxn runtime compilation with eigen headers:
https://github.com/microsoft/onnxruntime/blob/v0.5.1/onnxruntime/core/util/eigen_common_wrapper.h#L16

Reply:
Please find where the warning comes from and fix the like eigen_common_wrapper.h

---
In reply to: [339512521](https://github.com/microsoft/onnxruntime/pull/2258#discussion_r339512521) [](ancestors = 339512521)

Reply:
eigen_common_wrapper.h and alike are for ONNX Runtime compilation which works just fine.
This patch address eigen external component compilation and works fine targeting an unused parameter specifically. If you like we can ignore all unused parameter warnings. In case you already ignore these warnings for gcc x86 please point us to the code so that we can extend it for Arm cross-toolchains without duplicating code.

Reply:
I'd prefer to disable the warning only for ARM specific code, in cmake files.

Reply:
issue is having patch is we run risk of conflict if we need to update eigen.  perhaps localize this change to when building ACL only.

Reply:
I totally agree with @jywu-msft

Reply:
Let us take a closer look Monday, the reason might be that we have used DEV_MODE=ON configuration which is also documented.

Reply:
Hi @mikecaraman , George Wu already sent out a fix.  https://github.com/microsoft/onnxruntime/commit/8a102c6e99555c22be63a8c754a65c63b501d5c5